pallets/liquidity-pools-gateway/src/lib.rs

```diff
@@ -208,6 +223,33 @@ pub mod pallet {
 	pub type DomainHookAddress<T: Config> =
 		StorageMap<_, Blake2_128Concat, Domain, [u8; 20], OptionQuery>;
 
+	/// Storage for routers.
+	///
+	/// This can only be set by an admin.
+	#[pallet::storage]
+	#[pallet::getter(fn routers)]
+	pub type Routers<T: Config> = StorageMap<_, Blake2_128Concat, T::Hash, T::Router>;
+
+	/// Storage for domain multi-routers.
+	///
+	/// This can only be set by an admin.
+	#[pallet::storage]
+	#[pallet::getter(fn domain_multi_routers)]
+	pub type DomainMultiRouters<T: Config> =
+		StorageMap<_, Blake2_128Concat, Domain, BoundedVec<T::Hash, T::MultiRouterCount>>;
+
+	/// Storage that keeps track of incoming message proofs.
+	#[pallet::storage]
+	#[pallet::getter(fn inbound_message_proof_count)]
+	pub type InboundMessageProofCount<T: Config> =
+		StorageMap<_, Blake2_128Concat, Proof, u32, ValueQuery>;
+
+	/// Storage that keeps track of incoming messages and the expected proof
+	/// count.
+	#[pallet::storage]
+	#[pallet::getter(fn inbound_messages)]
+	pub type InboundMessages<T: Config> = StorageMap<_, Blake2_128Concat, Proof, (T::Message, u32)>;
```

Comment:
And when a message comes from a router, it will add a vote automatically. That way you can process the message when votes from all are higher than one. 

Not sure I'm explaining this well

Reply:
For inbound messages, we do not have a way of identifying routers, we just get a message from the queue that has a domain address and the actual message. The router hashes are only used in the context of outbound messages.

Reply:
That's correct, but why then the Solidity part is as it is (?). Maybe we need to specify a new parameter in the axelar part that carries that info?